### PR TITLE
Feature/event type time restrictions

### PR DESCRIPTION
**Pull request by romanetar**
ref https://tipit.avaza.com/project/view#!tab=task-pane&groupby=MyTaskDueDate&view=vertical&task=3325733&stafilter=NotComplete&fileview=grid

Signed-off-by: romanetar <roman_ag@hotmail.com>

### CODE REVIEW

File: app/Models/Foundation/Summit/TimeDurationRestrictedEvent.php [hunk 1]
```diff
@@ -87,6 +88,18 @@ private function _setDuration(?Summit $summit, int $duration_in_seconds, bool $s
             throw new ValidationException(sprintf('Duration should be greater than %s minutes.', self::MIN_EVENT_MINUTES));
         }
 
+        $type = $this->getType();
+        if ($type instanceof PresentationType) {
```
Comment (smarcet) [inline app/Models/Foundation/Summit/TimeDurationRestrictedEvent.php:92]: but this is not having in consideration the fact that for admins this restriction has no meaning, its only meant for track chairs

**Review by smarcet — CHANGES_REQUESTED** [review]
@romanetar please review comments

File: app/Models/Foundation/Summit/TimeDurationRestrictedEvent.php (hunk 1 above)
Comment (romanetar) [reply app/Models/Foundation/Summit/TimeDurationRestrictedEvent.php]: done

File: app/Models/Foundation/Summit/TimeDurationRestrictedEvent.php [hunk 2]
```diff
@@ -88,15 +89,19 @@ private function _setDuration(?Summit $summit, int $duration_in_seconds, bool $s
...
-        $type = $this->getType();
-        if ($type instanceof PresentationType) {
-            $min_duration = $type->getMinDuration();
-            if ($min_duration > 0 && $duration_in_seconds < ($min_duration * 60)) {
-                throw new ValidationException(sprintf('Duration must be greater than or equal to %s minutes.', $min_duration));
-            }
-            $max_duration = $type->getMaxDuration();
-            if ($max_duration > 0 && $duration_in_seconds > ($max_duration * 60)) {
-                throw new ValidationException(sprintf('Duration must be less than or equal to %s minutes.', $max_duration));
+        $isAdmin = !is_null($member) && ($member->isAdmin() || $member->isSummitAdmin());
```
Comment (smarcet) [inline app/Models/Foundation/Summit/TimeDurationRestrictedEvent.php:92]: just check Member::isSummitAllowed

**Review by smarcet — APPROVED** [review]
LGTM

**Pull request merged**